sklearn/utils/_metadata_requests.py

```diff
@@ -1385,28 +1385,74 @@ def __init_subclass__(cls, **kwargs):
         .. [1] https://www.python.org/dev/peps/pep-0487
         """
         try:
-            requests = cls._get_default_requests()
+            for method in SIMPLE_METHODS:
+                method_metadata_args = cls._get_metadata_args_and_aliases(method)
```

Comment:
instead of building the whole default request object, we now only check for parameters which are to be included here in `__init_subclass__`, which is what we actually need. The actual `MetadataRequest` object is now only created when needed.